### PR TITLE
Adding prometheus flags to make its configuration similar to the managed prometheus.

### PR DESCRIPTION
**Pull request by saketjajoo**
This commit adds some extra flags to the prometheus's deployment so that its configuration is similar to the managed prometheus' configuration.
Reference: https://github.com/GoogleCloudPlatform/prometheus-engine/blob/main/cmd/operator/deploy/operator/10-collector.yaml#L93-L109

### CODE REVIEW

File: prombench/manifests/prombench/benchmark/3b_prometheus-test_deployment.yaml
Comment (macxamin) [inline prombench/manifests/prombench/benchmark/3b_prometheus-test_deployment.yaml:179]: why did you add `--export.user-agent-mode=kubectl` 

https://github.com/GoogleCloudPlatform/prometheus-engine/blob/722ee81b87348557d7274be0dccfa14c3987b9af/cmd/operator/deploy/operator/10-collector.yaml#L109C11-L109C43

File: prombench/manifests/prombench/benchmark/3b_prometheus-test_deployment.yaml
Comment (macxamin) [inline prombench/manifests/prombench/benchmark/3b_prometheus-test_deployment.yaml:167]: same comment as pr-prometheus for args

File: prombench/manifests/prombench/benchmark/3b_prometheus-test_deployment.yaml
Comment (macxamin) [inline prombench/manifests/prombench/benchmark/3b_prometheus-test_deployment.yaml:69]: Nit: Can you format it differently so future maintainers can more easily understand  e.g.
```
# default prombench args
          "--web.external-url=http://{{ .DOMAIN_NAME }}/{{ .PR_NUMBER }}/prometheus-release",
          "--storage.tsdb.path=/prometheus",
          "--web.console.libraries=/etc/prometheus/console_libraries",
          "--web.console.templates=/etc/prometheus/consoles",
          "--config.file=/etc/prometheus/prometheus.yml",
        # JSON log format is needed for GKE to display log levels correctly.
        "--log.format=json",
        "--log.level=debug",
# following args are copied from https://github.com/GoogleCloudPlatform/prometheus-engine/blob/722ee81b87348557d7274be0dccfa14c3987b9af/cmd/operator/deploy/operator/10-collector.yaml#L92
# ensure both are in sync, exclude flags that are not relevant.
        - --enable-feature=exemplar-storage
        - --storage.tsdb.no-lockfile
        # Keep 30 minutes of data. As we are backed by an emptyDir volume, this will count towards
        # the containers memory usage. We could lower it further if this becomes problematic, but
        # it the window for local data is quite convenient for debugging.
        - --storage.tsdb.retention.time=30m
        - --storage.tsdb.wal-compression
        # Effectively disable compaction and make blocks short enough so that our retention window
        # can be kept in practice.
        - --storage.tsdb.min-block-duration=10m
        - --storage.tsdb.max-block-duration=10m
        - --export.user-agent-mode=kubectl
```

**Review by macxamin — COMMENTED** [review]
Good job, a few readability changes and it's good to go.

**Comment by macxamin** [comment]
@bwplotka any other configs that should be copied over?

**Review by macxamin — APPROVED** [review]
Great work! Merge after @bwplotka  approves.

File: prombench/manifests/prombench/benchmark/3b_prometheus-test_deployment.yaml
Comment (bwplotka) [inline prombench/manifests/prombench/benchmark/3b_prometheus-test_deployment.yaml:75]: Would be lovely to maybe parametrized the upstream - but we can do that later, LGTM for now.

Comment (bwplotka) [reply prombench/manifests/prombench/benchmark/3b_prometheus-test_deployment.yaml:75]: Do you mind create issue on this repo about it?

**Review by bwplotka — APPROVED** [review]
LGTM, just one ask (:

File: prombench/manifests/prombench/benchmark/3b_prometheus-test_deployment.yaml
Comment (macxamin) [reply prombench/manifests/prombench/benchmark/3b_prometheus-test_deployment.yaml:75]: @TheSpiritXIII can you add your thoughts about using Kustomize to have a sync between prometheus-engine collector config and the prombench collector configs here.

**Pull request merged**
